version: "3"
services:
  neo4j:
    image: neo4j:5.13.0
    volumes:
      - ./data:/data
      - ./plugins:/plugins
      - ./import:/import
    ports:
      - "7474:7474"
      - "7687:7687"
    expose:
      - 7474
      - 7687
    environment:
      - NEO4J_server_default__listen__address=0.0.0.0
      - NEO4J_initial_dbms_default__database=metro
      - NEO4J_AUTH=neo4j/BDII2023
      - NEO4J_dbms_security_procedures_unrestricted=apoc*,gds*
    user: "${UID}:${GID}"
  notebook:
    image: jupyter/datascience-notebook
    user: root
    volumes:
      - ./sources:/home/${USER}/work
      - ./startup:/usr/local/bin/before-notebook.d/
    ports:
      - "${JUPYTER_PORT}:8888"
    expose:
      - ${JUPYTER_PORT}
    environment:
      - GRANT_SUDO=1
      - CHOWN_HOME=yes
      - NB_USER=${USER}
      - NB_UID=${UID}
      - NB_GID=${GID}
